src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/RewriteMapParser.cs

```diff
@@ -0,0 +1,31 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace Microsoft.AspNetCore.Rewrite.Internal.IISUrlRewrite
+{
+    public class RewriteMapParser
+    {
+        public IDictionary<string, IISRewriteMap> Parse(XElement xmlRoot)
+        {
+            var mapsElement = xmlRoot.Descendants(RewriteTags.RewriteMaps).SingleOrDefault();
+            if (mapsElement == null)
+            {
+                return null;
+            }
+
+            var rewriteMaps = new Dictionary<string, IISRewriteMap>();
+            foreach (XElement mapElement in mapsElement.Elements(RewriteTags.RewriteMap))
+            {
+                var map = new IISRewriteMap(mapElement.Attribute(RewriteTags.Name)?.Value);
```

Comment:
Don't think this should be nullable; if the map doesn't have a name, we should probably throw.